subsys/net/lib/lwm2m_client_utils/Kconfig

```diff
@@ -57,6 +57,42 @@ config LWM2M_CLIENT_UTILS_DOWNLOADER_SEC_TAG
 	  https:// or coaps:// URI. It has no effect on usage of http:// or coap://
 	  -1 disables the usage of security tag.
 
+config LWM2M_CLIENT_UTILS_FIRMWARE_INSTANCE_MODEM
+	int "Modem Firmware Object instance ID, highest in the range is not usable"
+	range 0 LWM2M_FIRMWARE_INSTANCE_COUNT
```

Comment:
IMO if the range cannot be used reliably in this case, I'd rather remove it instead of elaborating in the help message why the range is not correct. Just specify in the help message that this value should be lower than `LWM2M_FIRMWARE_INSTANCE_COUNT` and verify this with the static asserts in the library.